### ISSUE TITLE
Magento 2.3.0: getSize call on configurable collection leads to exception, if no product filters are applied

### ISSUE THREAD

**Issue by jhruehl** [magento/magento2#22127]
### Preconditions (*)
1. Magento 2.3.0
2. NewRelic module enabled (and configured)
(you can cheat around enabling the module, if set the newrelic enabled setting to true in the core_config_table and flush the cache afterwards)

### Steps to reproduce (*)
1. Run magento_newrelicreporting_cron cron job

### Expected result (*)
1. magento_newrelicreporting_cron cron job successful

### Actual result (*)
1. magento_newrelicreporting_cron throws error "Warning: Invalid argument supplied for foreach() in /var/www/shop_test/src/www/vendor/magento/module-configurable-product/Model/ResourceModel/Product/Type/Configurable/Product/Collection.php on line 83"

### Explanation
The newrelic cron tries to get the amount of configurable products (or at least the given children) calling the getCount function of the ConfigurableProductManagementInterface.
In this process a configurable collection object is created (magento/module-configurable-product/Model/ResourceModel/Product/Type/Configurable/Product/Collection.php).
There the getSize function (in AbstractDB parent class) is called and as a consequence  the _renderFilters function in the magento/module-configurable-product/Model/ResourceModel/Product/Type/Configurable/Product/Collection.php class.

Since 2.3.0 the product filter functionality seems to have been partly moved there with the probable intention to enable several product filters to be applicable at the same time.

Here the _renderFilters() function code:

```
protected function _renderFilters()
    {
        parent::_renderFilters();
        $metadata = $this->getProductEntityMetadata();
        $parentIds = [];
        foreach ($this->products as $product) {
            $parentIds[] = $product->getData($metadata->getLinkField());
        }

        $this->getSelect()->where('link_table.parent_id in (?)', $parentIds);

        return $this;
    }
```

The cause of the problem should be quite clear, if you look at the code. If no product filter is applied to the configurable product collection $this-products is null, but without any checks it is used in the given foreach-loop.

Proposed fix:

```
protected function _renderFilters()
    {
        parent::_renderFilters();
        if ($this->products) {
            $metadata = $this->getProductEntityMetadata();
            $parentIds = [];
            foreach ($this->products as $product) {
                $parentIds[] = $product->getData($metadata->getLinkField());
            }
            $this->getSelect()->where('link_table.parent_id in (?)', $parentIds);
        }
        return $this;
    }
```

A simple and reasonable if-clause does the trick without touching the changed (or "enhanced") product filter functionality.

**Comment by jhruehl** [comment]
Sadly the insert-code functionality here doesn't do what I wanted or does strange stuff, but I think it should be still clear.

**Comment by jhruehl** [comment]
Sorry. Don't have the time to reproduce in vanilla, but a developer, who looks at the currently given code, should see the problem with it anyway. Basically it is impossible to reproduce in vanilla, if no newrelic account is set up for it, except cheating around the enabling by manipulating the enabled-value in the core_config_table.

**Comment by orlangur** [comment]
@jhruehl this is just a bot :) Could you create a PR via GitHub UI maybe? Fix seems to be pretty trivial.

**Comment by jhruehl** [comment]
Sadly gitHub doesn't create my patch-branch despite stating so on the pull request creation (or giving me the option). So comitting it fails due to a non existing patch branch, which should be created automatically beforehand.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99052` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @jhruehl, @davidverholen. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. 

The fix will be available with the upcoming `2.3.3` release.
